metaflow/plugins/kfp/kfp_utils.py

```diff
@@ -0,0 +1,171 @@
+    sort_by: Can be format of “field_name”, “field_name asc” or “field_name desc”
+      (Example, “name asc” or “id desc”). Ascending by default.
+    verbose: Print pipeline info if True.
+      Default to True since this function is intended to be used interactively (human facing).
+      For similar reason `print` is used over logging.info to avoid silencing output by default
+      logging level.
+    """
+    client = _get_kfp_client()
+    versions = client.list_pipeline_versions(
+        pipeline_id=pipeline_id, sort_by=sort_by
+    ).versions
+
+    if verbose:
+        print(f"{len(versions)} versions found for pipeline {pipeline_id}.")
+        print(f"Versions sorted by {sort_by}.")
+
+        for version_count, version in enumerate(versions, start=1):
+            print(f"\n=== Version {version_count} ===")
+            for key, value in version.to_dict().items():
+                print(f"{key}: {value}")
+
+    return versions
+
+
+def trigger_flow(
+    pipeline_name: str,
+    triggerred_flow_name: str,
+    namespace: str,
+    experiment_name: str = None,
+    pipeline_parameters: dict = None,
+    pipeline_version: str = None,
+) -> str:
+    """Trigger KFP flow by pipeline name. See trigger_flow_by_id for more details."""
+
+    client = _get_kfp_client()
+
+    pipeline_id = client.get_pipeline_id(name=pipeline_name)
+    return trigger_flow_by_id(
+        pipeline_id=pipeline_id,
+        experiment_name=experiment_name,
+        namespace=namespace,
+        triggerred_flow_name=triggerred_flow_name,
+        pipeline_parameters=pipeline_parameters,
+        pipeline_version=pipeline_version,
+    )
+
+
+def trigger_flow_by_id(
+    pipeline_id: str,
+    triggerred_flow_name: str,
+    namespace: str,
+    experiment_name: str = None,
+    pipeline_parameters: dict = None,
+    pipeline_version: str = None,
+) -> str:
+    """Trigger KFP flow by pipeline id.
+    pipeline_id: Pipeline id for which a new run should be triggered.
+    experiment_name: Experiment where the triggered run should be placed.
+    job_name: Job name of the flow being triggered.
+
+    Return run id of created run. To reconstruct url see run_id_to_url function.
+    """
+    client = _get_kfp_client()
+
+    if pipeline_parameters is None:
+        pipeline_parameters = {}
+
+    # Not checking for experiment existence: kfp client does not recreate experiment if exists
+    experiment: kfp_server_api.ApiExperiment = client.create_experiment(
+        name=experiment_name,
+        description="Experiment flow trigger from same namespace",
+        namespace=namespace,  # TODO: Warn about permission issue early
+    )
+
+    pipeline_run: kfp_server_api.ApiRun = client.run_pipeline(
+        experiment_id=experiment.id,
+        job_name=triggerred_flow_name,
+        pipeline_id=pipeline_id,
+        params={"flow_parameters_json": json.dumps(pipeline_parameters)},
+        version_id=pipeline_version,
+    )
+
+    return pipeline_run.id
```

Comment:
I would return the `pipeline_run` object instead of just the `.id` attribute as it gives the customer most flexibility on what to do with this information.